roles/network_plugin/calico/templates/cni-calico.conflist.j2

```diff
@@ -3,7 +3,7 @@
   "cniVersion":"0.3.1",
   "plugins":[
     {
-    {% if cloud_provider is defined %}
+    {% if cloud_provider is defined and cloud_provider != "openstack" %}
```

Comment:
This is definitely not the right answer. 
```
- name: Calico | Get kubelet hostname
  shell: >-
    {{ bin_dir }}/kubectl get node -o custom-columns='NAME:.metadata.name,INTERNAL-IP:.status.addresses[?(@.type=="InternalIP")].address'
    | egrep "{{ ansible_all_ipv4_addresses | join('$|') }}$" | cut -d" " -f1
  register: calico_kubelet_name
  delegate_to: "{{ groups['kube-master'][0] }}"
  when: cloud_provider is defined
```
^ This runs when using cloud_provider. All of them, including openstack. Nodes won't even register otherwise. Fix the command if it doesn't work in your env, but it's totally fine in mine:
```
ubuntu@control-01:~$ kubectl get node -o custom-columns='NAME:.metadata.name,INTERNAL-IP:.status.addresses[?(@.type=="InternalIP")].address'
NAME                                     INTERNAL-IP
control-01.pyrex.mosesohn01   10.232.255.5
control-02.pyrex.mosesohn01   10.232.255.7
control-03.pyrex.mosesohn01   10.232.255.8
worker-02.pyrex.mosesohn01    10.232.255.9
```